src/pages/home/report/ReportActionCompose/ReportActionCompose.js

```diff
@@ -342,121 +343,123 @@ function ReportActionCompose({
-                                    handleSendMessage={handleSendMessage}
-                                    shouldShowComposeInput={shouldShowComposeInput}
-                                    onFocus={onFocus}
-                                    onBlur={onBlur}
-                                    measureParentContainer={measureContainer}
-                                    listHeight={listHeight}
-                                />
-                                <ReportDropUI
-                                    onDrop={(e) => {
-                                        if (isAttachmentPreviewActive) {
-                                            return;
-                                        }
-                                        const data = lodashGet(e, ['dataTransfer', 'items', 0]);
-                                        displayFileInModal(data);
-                                    }}
-                                />
-                            </>
+                        <AttachmentModal
+                            headerTitle={translate('reportActionCompose.sendAttachment')}
+                            onConfirm={addAttachment}
+                            onModalShow={() => setIsAttachmentPreviewActive(true)}
+                            onModalHide={onAttachmentPreviewClose}
+                        >
+                            {({displayFileInModal}) => (
+                                <>
+                                    <AttachmentPickerWithMenuItems
+                                        displayFileInModal={displayFileInModal}
+                                        reportID={reportID}
+                                        report={report}
+                                        reportParticipantIDs={reportParticipantIDs}
+                                        isFullComposerAvailable={isFullComposerAvailable}
+                                        isComposerFullSize={isComposerFullSize}
+                                        updateShouldShowSuggestionMenuToFalse={updateShouldShowSuggestionMenuToFalse}
+                                        isBlockedFromConcierge={isBlockedFromConcierge}
+                                        disabled={disabled}
+                                        setMenuVisibility={setMenuVisibility}
+                                        isMenuVisible={isMenuVisible}
+                                        onTriggerAttachmentPicker={onTriggerAttachmentPicker}
+                                        onCanceledAttachmentPicker={restoreKeyboardState}
+                                        onMenuClosed={restoreKeyboardState}
+                                        onAddActionPressed={onAddActionPressed}
+                                        onItemSelected={onItemSelected}
+                                        actionButtonRef={actionButtonRef}
+                                    />
+                                    <ComposerWithSuggestions
```

Comment:
Looks like that `PersonalDetailsContext` is useful for `ComposerWithSuggestions` only as we are using this in `SuggestionMention`. Can you move the `PersonalDetailsContext` wrapper below the tree to only wrap `ComposerWithSuggestions` ?

Reply:
@hurali97 Thanks for the idea! During yesterday's investigations of different issue, I discovered that we can use `withPersonalDetails` HOC that has cache layer (in form of a React context) between Onyx and UI. This will have the same impact on performance but we wouldn't need to create a separate context. Please check out the changes now - much cleaner 👍

Reply:
@hurali97 there is an issue with existing withPersonalDetails HOC, so I will remove those improvement and just bring the context down.

Reply:
Thank you for clarifying what you meant, it's ready now ✅ 😄

Reply:
Cool !!